packages/admin/admin/src/form/FinalFormChooseFile.tsx

```diff
@@ -114,27 +128,41 @@ const styles = ({ palette }: Theme) => {
     });
 };
 
-export interface FinalFormFileUploadProps extends FieldRenderProps<File | File[], HTMLInputElement> {
+export interface FinalFormChooseFileFieldProps extends FieldRenderProps<File | File[], HTMLInputElement> {
     disableDropzone?: boolean;
-    disableButton?: boolean;
+    disableSelectFileButton?: boolean;
     accept: Accept;
     maxSize: number;
+    maxFiles?: number;
+    iconMapping?: {
+        delete?: React.ReactNode;
+        info?: React.ReactNode;
+        select?: React.ReactNode;
+    };
 }
 
-const FinalFormFileUploadComponent: React.FunctionComponent<WithStyles<typeof styles> & FinalFormFileUploadProps> = ({
+const FinalFormChooseFileFieldComponent: React.FunctionComponent<WithStyles<typeof styles> & FinalFormChooseFileFieldProps> = ({
     classes,
     disabled,
     disableDropzone,
-    disableButton,
+    disableSelectFileButton,
     accept,
     maxSize = 50 * 1024 * 1024,
+    maxFiles,
     input: { onChange, value: fieldValue, multiple: multipleFiles },
+    iconMapping = {},
 }) => {
+    const { delete: deleteIcon = <Delete />, info: infoIcon = <Info />, select: selectIcon = <Select /> } = iconMapping;
+
     const onDrop = React.useCallback(
         (acceptedFiles: File[]) => {
-            multipleFiles && Array.isArray(fieldValue) ? onChange([...fieldValue, ...acceptedFiles]) : onChange([...acceptedFiles]);
+            multipleFiles && Array.isArray(fieldValue)
+                ? maxFiles && fieldValue.length < maxFiles && fieldValue.length + acceptedFiles.length <= maxFiles
+                    ? onChange([...fieldValue, ...acceptedFiles])
+                    : onChange([...acceptedFiles])
+                : onChange([...acceptedFiles]);
```

Comment:
This condition is difficult to comprehend. Could you please make it easier to read by using if instead of ternary operators?